### PR TITLE
[Regression Test] Add sage neighbor sample test for reddit

### PR DESCRIPTION
**Pull request by classicsong**
## Description
Add speed test for sage neighbor sample for reddit and ogbn-products

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

**Comment by VoVAllen** [comment]
Have you tested on cpu & gpu?

**Comment by classicsong** [comment]
> Have you tested on cpu & gpu?

Yes

File: benchmarks/benchmarks/utils.py [hunk 1]
```diff
@@ -37,11 +37,61 @@ def get_graph(name):
         print(name + " doesn't exist")
         return None
 
+class ogb_data(object):
```
Comment (jermainewang) [inline benchmarks/benchmarks/utils.py:40]: `OGBDataSet`

**Review by jermainewang — APPROVED** [review]
LGTM

**Review by VoVAllen — APPROVED** [review]
LGTM

File: benchmarks/benchmarks/utils.py [hunk 2]
```diff
@@ -37,11 +37,61 @@ def get_graph(name):
...
+    def num_classes(self):
+        return self._num_labels
+
+    def __getitem__(self, idx):
+        return self._g
+
+def load_ogb_product(name):
+    from ogb.nodeproppred import DglNodePropPredDataset
+
+    os.symlink('/tmp/dataset/', os.path.join(os.getcwd(), 'dataset'))
```
Comment (VoVAllen) [inline benchmarks/benchmarks/utils.py:59]: Change it to an environment variable or something like `~/.dgl/ogb/`?

Comment (classicsong) [reply benchmarks/benchmarks/utils.py:59]: You can make the decision and let me know:)

**Pull request merged**